src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.ErrorTypeSymbolKey.cs

```diff
@@ -77,15 +83,12 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
 
                 if (typeArgumentsFailureReason != null)
                 {
+                    Contract.ThrowIfFalse(typeArguments.IsDefault);
                     failureReason = $"({nameof(ErrorTypeSymbolKey)} {nameof(typeArguments)} failed -> {typeArgumentsFailureReason})";
                     return default;
                 }
 
-                if (typeArguments.IsDefault)
-                {
-                    failureReason = $"({nameof(ErrorTypeSymbolKey)} {nameof(typeArguments)} failed)";
-                    return default;
-                }
```

Comment:
this check is redundant as per the rules of ReadSymbolKeyArray (now fully documented).